### ISSUE TITLE
Evaluate module during fine-tuning

### ISSUE THREAD

**Issue by gorjanradevski** [tensorflow/hub#336]
I am trying to fine-tune a Tensorflow Hub module and evaluate it on the validation set during training. To be more specific, I want to import the module in train mode, train it for an epoch, evaluate the performance of the whole model (includes other trainable layers) on the validation set, then train it again...
 
In order to achieve that, here is a pseudo code of what I am trying to do:
```
is_training = tf.placeholder_with_default(False, None)
resnet_test = hub.Module( "https://tfhub.dev/google/imagenet/resnet_v2_152/feature_vector/3")
resnet_train = hub.Module("https://tfhub.dev/google/imagenet/resnet_v2_152/feature_vector/3", trainable=True, tags={"train"})

features = tf.cond(is_training, true_fn=lambda: resnet_train(images, signature="image_feature_vector_with_bn_hparams"), false_fn=lambda: resnet_test(images, signature="image_feature_vector")

dense = tf.layers.dense(features, 10)

....

with tf.Session() as sess:
     sess.run(train_step, feed_dict={is_training:True})
     sess.run(val_step)
```
However, by doing this the Tensorflow graph contains two modules, namely `module` and `module_2`. On the other hand, I want to have two modules, that will share their weights, where one will be imported in train mode and the other in evaluation mode.

Also, is there a simpler way to train a module and evaluate it that I am not aware of?

I also asked a [Stackoverflow question](https://stackoverflow.com/questions/57240336/tensorflow-hub-evaluate-module-during-fine-tuning) for the raised issue.

Looking forward to you help!

**Comment by gorjanradevski** [comment]
In the meantime, I may have found a way around the issue so I will post it here as an optional solution. Let me know of your thoughts:

    is_training = tf.placeholder_with_default(False, None)
    resnet = hub.Module("https://tfhub.dev/google/imagenet/resnet_v2_152/feature_vector/3", trainable=True, tags={"train"})
    signature = tf.cond(is_training, true_fn=lambda: "image_feature_vector", false_fn=lambda: "image_feature_vector_with_bn_hparams")
    output = resnet(image)

Then, if I assert the outputs I am getting when `is_training` is set to `True` and `is_training` is set to `False` are indeed different, but I can't validate for sure if they are correct.

**Comment by arnoegw** [comment]
Thanks for highlighting the cross-posting! I've send a comprehensive answer on [StackOverflow](https://stackoverflow.com/questions/57117039/tensorflow-hub-fine-tune-and-evaluate). Its TL;DR is: Your analysis is correct, and the problem will be fixed with the TF2 flavor for TF Hub. However, we do not intend to go back and change things for TF1.

With that, is it fair to close this issue?

**Comment by gorjanradevski** [comment]
Thanks Arno, feel free to close the issue here. However, if you have the time, I asked one more question in the comment section on [Stackoverflow](https://stackoverflow.com/questions/57117039/tensorflow-hub-fine-tune-and-evaluate).
